packages/vsx-registry/src/node/vsx-extension-resolver.ts

```diff
@@ -43,23 +44,38 @@ export class VSXExtensionResolver implements PluginDeployerResolver {
         return !!VSXExtensionUri.toId(new URI(pluginId));
     }
 
-    async resolve(context: PluginDeployerResolverContext): Promise<void> {
+    async resolve(context: PluginDeployerResolverContext, options?: PluginDeployOptions): Promise<void> {
         const id = VSXExtensionUri.toId(new URI(context.getOriginId()));
         if (!id) {
             return;
         }
-        console.log(`[${id}]: trying to resolve latest version...`);
-        const extension = await this.api.getLatestCompatibleExtensionVersion(id);
-        if (!extension) {
-            return;
-        }
-        if (extension.error) {
-            throw new Error(extension.error);
+
+        let extensionVersion: string | undefined;
+        let extension: VSXExtensionRaw | undefined;
+        if (options) {
+            extensionVersion = options.version;
+            console.log(`[${id}]: trying to resolve version ${extensionVersion}...`);
+            extension = await this.api.getExtension(id, { extensionVersion: extensionVersion });
+            if (!extension) {
```

Comment:
The error handling is duplicated here. Could we have the `if (options)` block only cover down to `extension = await ...`  code, and then run the error handling?